File: 14. Longest Common Prefix/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size()==0) return "";
        int minLength = INT_MAX;
        string res = "";
        for(int i=0;i<strs.size();i++){
            if(minLength>strs[i].length()) minLength = strs[i].length();
        }
        for(int i=0;i<minLength;i++){
            char start = strs[0][i];
            for(int j=0;j<strs.size();j++){
                if(strs[j][i]!=start){
                    // cout << strs[i][j] << start;
                    return res;
                }
            }
            res += start;
        }
        return res;
    }
};

File: 14. Longest Common Prefix/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minLength = Integer.MAX_VALUE;
        for(int i=0;i<strs.length;i++){
            minLength = Math.min(minLength,strs[i].length());
        }
        if(minLength==Integer.MAX_VALUE){
            return("");
        }
        String res="";
        for(int j=0;j<minLength;j++){
            String start = strs[0].substring(j,j+1);
            for(int i=1;i<strs.length;i++){
                if(!strs[i].substring(j,j+1).equals(start)){
                    return res;
                }
            }
            res += start;
        }
        return(res);
    }
}